src/infra/postgres/client/client.go

```diff
@@ -120,3 +130,20 @@ func (c PostgresClient) RunInTransaction(ctx context.Context, persist func(clien
 
 	return c.db.(*pg.DB).RunInTransaction(ctx, persistFunc)
 }
+
+func (c *PostgresClient) SelectMany(ctx context.Context, model, dst interface{}, condition string, params ...interface{}) error {
```

Comment:
More than a `SelectMany` I would rather name this SelectWithQuery with the following impl:
```
func (c *PostgresClient) SelectWithQuery(ctx context.Context, model interface{}, *pg.Query) {
        q := c.db.QueryContext(ctx, model, query).Select()
	if err != nil {
		return err
	}
	return nil
}
```

Do you think that can make the work?

Reply:
It couples the Store interface to the `go-pg` library, which would break the point of having the interface in my opinion.
I could rename the func, though. I'm not very happy with the naming myself.
Also, I think Dario created another func with many rows return. I need to check it and update my code.

Reply:
I renamed it `SelectWhere` and reduced the number of args, without depending on `pg` specific types.